thread.go

```diff
@@ -88,11 +88,21 @@ const (
 )
 
 type ThreadMessage struct {
-	Role        ThreadMessageRole  `json:"role"`
-	Content     string             `json:"content"`
-	FileIDs     []string           `json:"file_ids,omitempty"`
-	Attachments []ThreadAttachment `json:"attachments,omitempty"`
-	Metadata    map[string]any     `json:"metadata,omitempty"`
+	Role        ThreadMessageRole      `json:"role"`
+	Content     []ThreadMessageContent `json:"content"`
```

Comment:
Hey, thank you for the PR!

The documentation says it is either a `string` or `array`:  https://platform.openai.com/docs/api-reference/messages/createMessage#messages-createmessage-content

So I think we need a new type here that supports proper marshalling

Reply:
Could we send an array of one with `text: { ... }` instead of a new type?